cf-pubsub/src/main/java/org/eclipse/californium/pubsub/PubSub.java

```diff
@@ -0,0 +1,343 @@
+     * @throws IOException if any other issue (not specific to the connector) occurred
+     */
+    public CoapResponse remove(String... uri) throws ConnectorException, IOException {
+
+        CoapClient client = new CoapClient(SCHEME, this.getHost(), this.getPort(), uri);
+        client.setTimeout(this.timeout);
+        CoapResponse res = client.delete();
+
+        return res;
+    }
+
+
+    /**
+     * The Subscription class can be used to asynchronously subscribe to a given topic uri
+     * It can be reused for the same uri while also giving it a new {@link CoapHandler} to handle the
+     * asynchronous CoapResponse returned from the broker
+     *
+     * The uri cannot be changed once set, but the CoapHanlder can be.
+     */
+    public class Subscription {
+        /** the client */
+        private CoapClient client;
+        /** the observe relation */
+        private CoapObserveRelation relation;
+        /** the uri of subscribed topic */
+        private String[] uri;
+        /** the listener for returned CoapResponse */
+        private CoapHandler handler;
+
+        /**
+         * Constructs a A Subscription instance that allows subscription to a given
+         * topic uri string and takes a a CoapHandler to handle the returned response
+         *
+         * A Constructed Susbcription instance does not automatically subscribe
+         * you must call #subscribe().
+         *
+         * @param handler CoapHandler
+         * @param uri String or String[]
+         */
+        public Subscription(CoapHandler handler, String... uri) {
+            this.uri = uri;
+            this.handler = handler;
+            this.relation = null;
+            this.client = null;
+        }
+
+        /**
+         * Gets the current handler assigned to Subscription instance
+         *
+         * @return a CoapHandler
+         */
+        public CoapHandler getHandler() {
+            return handler;
+        }
+
+        /**
+         * Sets a new CoapHandler for the Subscription instance
+         *
+         * @param handler CoapHandler
+         */
+        public void setHandler(CoapHandler handler) {
+            this.handler = handler;
+        }
+
+        /**
+         * subscribes to the uri of the Subscription instance and runs
+         * the CoapHandler #onLoad when it gets a CoapResponse
+         *
+         */
+        public void subscribe() {
+
+            Request req = new Request(CoAP.Code.GET);
+
+            client = new CoapClient(SCHEME, getHost(), getPort(), uri);
+            client.useExecutor();
+            client.setTimeout(timeout);
+
+            req.setURI(client.getURI());
+            req.setObserve();
+
+            config.set(NetworkConfig.Keys.TOKEN_SIZE_LIMIT, 4);
+            RandomTokenGenerator rand = new RandomTokenGenerator(config);
+            Token token = rand.createToken(false);
+            req.setToken(token);
```

Comment:
If you set TOKEN_SIZE_LIMIT in the "Standard" NetworkConfig (as done in line 321), it should be effective without generating a token on your own.